### PR TITLE
Refactor and fixes for making this importable

### PR DESCRIPTION
**Pull request by adwait**
- changed repo name (dropped the `-`)
- minor fixes and refactoring
- added Uext support

### CODE REVIEW

File: src/program.py [hunk 1]
```diff
@@ -60,7 +60,9 @@ def isin(self, p) -> bool:
         return False
 
     def serialize(self) -> str:
-        return str(self.lid) + " " + self.inst + " " + [str(op.lid) + " " for op in self.operands]
+        return str(self.lid) + " " + self.inst + " " + \
+            ' '.join([(str(op.lid) if isinstance(op, Instruction) else str(op)) + " " \
+                      for op in self.operands ])
```
Comment (dobios) [inline src/program.py:67]: isn't there a risk of printing garbage if the operand isn't an instruction? Like how can this guarantee that the output will be valid btor2? Wouldn't we want to sanitize a bit more and check if the param is an int if not an instruction?

File: src/passes/allpasses.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+from passes.transforms.initAllStates import *
+from passes.transforms.renameInputs import *
+from passes.validation.checkLidOrdering import *
+
+# Retrieves a pass from the list given an id
+def find_pass(p: list[Pass], id: str) -> Pass:
+    return next((e for e in p if e.id == id), None)
+
+# List containing all passes
+all_passes = [RenameInputs(), InitAllStates(), CheckLidOrdering()]
```
Comment (dobios) [inline src/passes/allpasses.py]: new line

File: src/passes/transforms/renameInputs.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    def run(p: list[Instruction]) -> list[Instruction]:
+        i = 0
+        res = []
+        for inst in p:
+            if isinstance(inst, Input):
+                res.append(Input(inst.lid, inst.sort, f"inp_{i}"))
+                i += 1
+            else:
+                res.append(inst)
+        return res
```
Comment (dobios) [inline src/passes/transforms/renameInputs.py]: new line

File: src/parse.py [hunk 4]
```diff
@@ -48,21 +51,23 @@ def parse(inp: list[str]) -> list[Instruction]:
                 # Construct instruction
                 op = Sort(lid, inst[2], int(inst[3]))
 
+
             case "input":
                 # Sanity check: verify that instruction is well formed
                 assert len(inst) >= 4,\
-                    "sort instruction must be of the form: <lid> input <sid> <name>. Found: " + line
+                    "input instruction must be of the form: <lid> input <sid> <name>. Found: " + line
```
Comment (dobios) [inline src/parse.py]: Thanks for catching my copy-pasta ^^"

File: __init__.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
+
+from .src import program
+from .src.parse import parse
+from .src.passes.passes import Pass
```
Comment (dobios) [inline __init__.py]: new line

Also you clearly know how to manage python projects better than me xD

File: src/program.py [hunk 6]
```diff
@@ -200,106 +210,117 @@ def serialize(self) -> str:
...
+class Uext(Instruction):
+    def __init__(self, lid: int, sort: Sort, op: Instruction, width: int, name: str):
+        super().__init__(lid, "uext", [sort, op, width, name])
+        # Handle btor uexts which are creating name aliases
+        self.width: int = width
+        self.renaming = False
+        if self.width == 0:
+            self.renaming = True
+            self.name = name
+            self.aliasid = op.lid
```
Comment (dobios) [inline src/program.py]: new line

Also lol I didn't know btro2 supported aliasing like this, I just forced myself to get rid of everything before emission in CIRCT. Nice

**Review by dobios — APPROVED** [review]
This is great!! Thanks so much for fixing this up !

**Comment by dobios** [comment]
> * changed repo name (dropped the `-`)
> 
>     * minor fixes and refactoring
> 
>     * added Uext support

Not sure changing the name is necessary, I was following the llvm compiler naming convention (llvm-opt. circt-opt, etc...). Given that this is currently purely a post-processing compiler meant to simply massage some btor2 with specific passes, I think the name is adequate.

**Comment by adwait** [comment]
Thanks for the comments, will fix them.

Agreed on renaming the repo, I did this since I just wanted to add this as a submodule (and wanted to have a python import which does not like `-`).

File: src/program.py (hunk 6 above)
Comment (adwait) [reply src/program.py]: This is really just something that yosys does when it wants to say that one wire (e.g., module port) is connected with another. 

Now that I check, it does seem to be a part of the btor2 spec though: https://fmv.jku.at/papers/NiemetzPreinerWolfBiere-CAV18.pdf

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
```
Comment (dobios) [inline .pre-commit-config.yaml:9]: wow fancy, thanks !!

File: tests/test.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    def test_btor1(self):
+        prgm = parse.parse(parsewrapper("tests/btor/reg_en.btor"))
+
+        self.assertEqual(prgm[0].inst, "sort")
+        self.assertEqual(prgm[1].inst, "input")
+        self.assertEqual(len(prgm), 22)
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (dobios) [inline tests/test.py:42]: You just saved me an afternoon of looking into how to write tests for python code, thanks this is really wonderful !!!

**Review by dobios — APPROVED** [review]

**Pull request merged**